### ISSUE TITLE
FLTK 1.4 - link errors connected with recent Zenity dialog feature

### ISSUE THREAD

**Issue by pdhahn** [fltk/fltk#602]
After today's FLTK 1.4 update, which included developments related to the Zenity file chooser (I know nothing about that), I cannot rebuild successfully. I get link errors on all test programs like the following...

```
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o): In function `Fl_Zenity_Native_File_Chooser_Driver::Fl_Zenity_Native_File_Chooser_Driver(int)':
Fl_Native_File_Chooser_Zenity.cxx:(.text+0x425): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::Fl_Kdialog_Native_File_Chooser_Driver(int)'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o): In function `Fl_Zenity_Native_File_Chooser_Driver::~Fl_Zenity_Native_File_Chooser_Driver()':
Fl_Native_File_Chooser_Zenity.cxx:(.text._ZN36Fl_Zenity_Native_File_Chooser_DriverD2Ev[_ZN36Fl_Zenity_Native_File_Chooser_DriverD5Ev]+0x8): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::~Fl_Kdialog_Native_File_Chooser_Driver()'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o): In function `Fl_Zenity_Native_File_Chooser_Driver::~Fl_Zenity_Native_File_Chooser_Driver()':
Fl_Native_File_Chooser_Zenity.cxx:(.text._ZN36Fl_Zenity_Native_File_Chooser_DriverD0Ev[_ZN36Fl_Zenity_Native_File_Chooser_DriverD5Ev]+0xc): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::~Fl_Kdialog_Native_File_Chooser_Driver()'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTI36Fl_Zenity_Native_File_Chooser_Driver[_ZTI36Fl_Zenity_Native_File_Chooser_Driver]+0x10): undefined reference to `typeinfo for Fl_Kdialog_Native_File_Chooser_Driver'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x40): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::count() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x48): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::filename() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x50): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::filename(int) const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x58): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::directory(char const*)'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x60): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::directory() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x68): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::title(char const*)'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x70): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::title() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x78): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::filter() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x80): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::filter(char const*)'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0x88): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::filters() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0xa0): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::preset_file(char const*)'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0xa8): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::preset_file() const'
../lib/libfltk.a(Fl_Native_File_Chooser_Zenity.cxx.o):(.rodata._ZTV36Fl_Zenity_Native_File_Chooser_Driver[_ZTV36Fl_Zenity_Native_File_Chooser_Driver]+0xb8): undefined reference to `Fl_Kdialog_Native_File_Chooser_Driver::show()'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/test/utf8] Error 1
make[2]: Target `test/CMakeFiles/utf8.dir/build' not remade because of errors.
make[1]: *** [test/CMakeFiles/utf8.dir/all] Error 2
```
According to the checkin notes, it looks like the Zenity stuff is somehow related to the KDialog, which option (OPTION_USE_KDIALOG) I have to disable in my cmake configuration. I have an old CentOS 6 system with g++ 5.1.

Does there need to be another cmake option flag to disable the Zenity stuff like with the KDialog stuff?

**Comment by pdhahn** [comment]
Oops somehow I missed the fact that you indeed added an OPTION_USE_ZENITY flag already.

Nevertheless, I have it "not selected" (just like with OPTION_USE_KDIALOG) in my cmake configuration and I still am getting these link errors.

**Comment by pdhahn** [comment]
Please fix his problem!

**Comment by pdhahn** [comment]
OK as a temp workaround, I put `#if USE_ZENITY` guards around contents of src/Fl_Native_File_Chooser_Zenity.cxx and around everything referring to Zenity in src/Fl_Native_File_Chooser_GTK.cxx. That seems to alleviate the problem in my local installation until you deal with it properly.

**Comment by ManoloFLTK** [comment]
@pdhahn  That should be fixed with commit b246b66. Please, close issue if you confirm.

**Comment by pdhahn** [comment]
Yes I believe the problem is fixed in the referenced commit. Thanks!
